[gn07] iP

gn07 opened the pull request:
# Grass 🌿

> Developers do not touch grass ~some non developer

[why touch grass?](https://www.techopedia.com/what-does-touching-grass-mean)

Has the following uses:
- developers get to *touch grass* without leaving their keyboards by developing on this project
- double use as a task manager

what we are currently improving on:
- [x] add a task manager
- [ ] improve grass quality

possible new future updates:
1. more grass
2. different variety of grass (for grass connoisseurs )

sneak peak on what java code looks like eg `main`:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

gn07 committed:
rename, greet and exit

gn07 committed:
Echo

gn07 committed:
Add, List

gn07 committed:
Mark as Done

gn07 committed:
ToDos, Events, Deadlines

gn07 committed:
Automated Text UI Testing

gn07 committed:
Handling Errors

gn07 committed:
Delete

gn07 committed:
A-Enums

gn07 committed:
add save to file feature

kevinlimantoro123 commented on src/main/java/Grass.java:
```diff
@@ -0,0 +1,314 @@
+import java.util.Scanner;
```
the import statement orderings are wrong

kevinlimantoro123 commented on src/main/java/Grass.java:
```diff
@@ -0,0 +1,314 @@
+        return;
+    }
+
+    public static void main(String[] args) {
```
This main function is really long. It would be better to abstract the command functions and put it in another folder

kevinlimantoro123 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,30 @@
+public class Task {
```
I like how clean this bit of code is!

kevinlimantoro123 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+public class Event extends Task {
```
Code is concise and clear!

kevinlimantoro123 reviewed (commented):
Overall, code is really clean and code. Maybe a bit more extraction of the code into different directories would be helpful

gn07 committed:
add dates and times

SherrinfordHR commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+public class Event extends Task {
+    protected String from;
```
Perhaps variable name "from" can be changed to startDate?

SherrinfordHR commented on src/main/java/Grass.java:
```diff
@@ -0,0 +1,308 @@
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+class GrassException extends Exception {
```
Perhaps put this class in a different java file?

gn07 committed:
make application use more OOP

gn07 committed:
A-packages

gn07 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

gn07 committed:
fix package errors

gn07 committed:
add gradle

gn07 committed:
add junit

gn07 committed:
generate jar file

gn07 committed:
add javadoc comments

gn07 committed:
updated coding standard for import statements

gn07 committed:
add find command

gn07 committed:
Merge branch 'codingStandard'

gn07 committed:
Merge branch 'heads/level-9'

gn07 committed:
add javadoc comment for parse find task

zlllllr commented on src/main/java/grass/Parser.java:
```diff
@@ -0,0 +1,193 @@
+package grass;
```
Perhaps you can put different classes into separate packages, such as putting Task, Todo, Deadline and Event into grass.task package

zlllllr commented on src/main/java/grass/Ui.java:
```diff
@@ -0,0 +1,230 @@
+* @author gn07
+* 
+*/
+public class Ui {
```
I like your Ui class which is very clear and logical.

zlllllr reviewed (commented):
LGTM! Overall, I think the code is clean and logical. Maybe reorganize the structure of packageing a little bit would make the code clearer. All the best to your 2103!

gn07 committed:
level 10-GUI

gn07 committed:
A-Assertions

Add assert statements to code

Some methods lack detailed checks for input values where assumptions on the input values are made.

Add assert statements to the methods.

This ensures application robustness by removing all assumptions in the code.

gn07 committed:
A-CodeQuality

Improve quality of code and comments

There were some improvements to be made on the quality of code and comments in the repo.

Removes unreachable lines, unused legacy code and ensure comments are updated to recent changes.

Improve code readability and understandability.

gn07 committed:
Merge pull request #2 from gn07/branch-A-Assertions

A-Assertions

gn07 committed:
Merge branch 'master' into branch-A-CodeQuality

gn07 committed:
Merge pull request #3 from gn07/branch-A-CodeQuality

A-CodeQuality

gn07 committed:
C-DetectDuplicates

gn07 committed:
update user guide and add title to Ui

gn07 committed:
Update README.md

gn07 committed:
Update Grass.java

gn07 committed:
Update README.md
